From a38c18624d82d41d71484b50ff072c18140aafa8 Mon Sep 17 00:00:00 2001
From: reichi <stephan@reichholf.net>
Date: Thu, 4 Jul 2019 12:57:34 +0200
Subject: [PATCH 1/3] fix system protobuf

---
 dependencies/CMakeLists.txt | 26 +++++++++++++-------------
 1 file changed, 13 insertions(+), 13 deletions(-)

diff --git a/dependencies/CMakeLists.txt b/dependencies/CMakeLists.txt
index fe3d463..34220f9 100644
--- a/dependencies/CMakeLists.txt
+++ b/dependencies/CMakeLists.txt
@@ -16,7 +16,7 @@ set(USE_SYSTEM_PROTO_LIBS OFF CACHE BOOL "use protobuf library from system")
 
 if (USE_SYSTEM_PROTO_LIBS)
 	find_package(Protobuf REQUIRED)
-	include_directories(${PROTOBUF_INCLUDE_DIRS})
+	include_directories(${Protobuf_INCLUDE_DIRS})
 else ()
 	set(BUILD_SHARED_LIBS OFF CACHE BOOL "Build shared protobuf library")
 	add_subdirectory(external/protobuf)
@@ -31,15 +31,15 @@ else ()
 	endif()
 
 	# define the include for the protobuf library at the parent scope
-	set(PROTOBUF_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/external/protobuf/src")
-	set(PROTOBUF_INCLUDE_DIRS ${PROTOBUF_INCLUDE_DIRS} PARENT_SCOPE)
+	set(Protobuf_INCLUDE_DIRS "${CMAKE_CURRENT_SOURCE_DIR}/external/protobuf/src")
+	set(Protobuf_INCLUDE_DIRS ${Protobuf_INCLUDE_DIRS} PARENT_SCOPE)
 
 	# define the protoc executable at the parent scope
-	get_property(PROTOBUF_PROTOC_EXECUTABLE TARGET protoc_compiler PROPERTY LOCATION)
-	set(PROTOBUF_PROTOC_EXECUTABLE ${PROTOBUF_PROTOC_EXECUTABLE} PARENT_SCOPE)
+	get_property(Protobuf_PROTOC_EXECUTABLE TARGET protoc_compiler PROPERTY LOCATION)
+	set(Protobuf_PROTOC_EXECUTABLE ${Protobuf_PROTOC_EXECUTABLE} PARENT_SCOPE)
 endif()
 
-message(STATUS "Using protobuf compiler: " ${PROTOBUF_PROTOC_EXECUTABLE})
+message(STATUS "Using protobuf compiler: " ${Protobuf_PROTOC_EXECUTABLE})
 
 #=============================================================================
 # Copyright 2009 Kitware, Inc.
@@ -55,13 +55,13 @@ message(STATUS "Using protobuf compiler: " ${PROTOBUF_PROTOC_EXECUTABLE})
 #=============================================================================
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
-function(PROTOBUF_GENERATE_CPP SRCS HDRS)
+function(Protobuf_GENERATE_CPP SRCS HDRS)
 	if(NOT ARGN)
-		message(SEND_ERROR "Error: PROTOBUF_GENERATE_CPP() called without any proto files")
+		message(SEND_ERROR "Error: Protobuf_GENERATE_CPP() called without any proto files")
 		return()
 	endif()
 
-	if(PROTOBUF_GENERATE_CPP_APPEND_PATH)
+	if(Protobuf_GENERATE_CPP_APPEND_PATH)
 		# Create an include path for each file specified
 		foreach(FIL ${ARGN})
 			get_filename_component(ABS_FIL ${FIL} ABSOLUTE)
@@ -75,8 +75,8 @@ function(PROTOBUF_GENERATE_CPP SRCS HDRS)
 		set(_protobuf_include_path -I ${CMAKE_CURRENT_SOURCE_DIR})
 	endif()
 
-	if(DEFINED PROTOBUF_IMPORT_DIRS)
-		foreach(DIR ${PROTOBUF_IMPORT_DIRS})
+	if(DEFINED Protobuf_IMPORT_DIRS)
+		foreach(DIR ${Protobuf_IMPORT_DIRS})
 			get_filename_component(ABS_PATH ${DIR} ABSOLUTE)
 			list(FIND _protobuf_include_path ${ABS_PATH} _contains_already)
 			if(${_contains_already} EQUAL -1)
@@ -86,7 +86,7 @@ function(PROTOBUF_GENERATE_CPP SRCS HDRS)
 	endif()
 
 	if(CMAKE_CROSSCOMPILING OR USE_SYSTEM_PROTO_LIBS)
-		set(PROTOC_DEPENDENCY ${PROTOBUF_PROTOC_EXECUTABLE})
+		set(PROTOC_DEPENDENCY ${Protobuf_PROTOC_EXECUTABLE})
 	else()
 		set(PROTOC_DEPENDENCY protoc_compiler)
 	endif()
@@ -103,7 +103,7 @@ function(PROTOBUF_GENERATE_CPP SRCS HDRS)
 		add_custom_command(
 			OUTPUT "${CMAKE_CURRENT_BINARY_DIR}/${FIL_WE}.pb.cc"
 				   "${CMAKE_CURRENT_BINARY_DIR}/${FIL_WE}.pb.h"
-			COMMAND ${PROTOBUF_PROTOC_EXECUTABLE}
+			COMMAND ${Protobuf_PROTOC_EXECUTABLE}
 			ARGS --cpp_out ${CMAKE_CURRENT_BINARY_DIR} ${_protobuf_include_path} ${ABS_FIL}
 			DEPENDS ${ABS_FIL} ${PROTOC_DEPENDENCY}
 			COMMENT "Running C++ protocol buffer compiler on ${FIL}"
-- 
2.17.1

